### DISCUSSION TITLE
Release 5.1.0 Deployment issues

### DISCUSSION THREAD

**Discussion by sandeep-r1** [Sunbird-Ed/Community#360] [Bugs]
**1)Nginx-public-ingress Deployment Issue**
We are trying to deploy Deploy/Kubernetes/nginx-public-ingress with public tag release-5.1.0_RC5 and facing the following error in jenkins. (jenkins job link attached in the weblinks)

 

`TASK [helm-deploy : rename template] *******************************************
fatal: [localhost]: FAILED! => {
    "changed": false
}

MSG:

AnsibleUndefinedVariable: list object has no element 2

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=1    skipped=1    rescued=0 `

**2)Content service Deployment.**
Content Editor deployment is job is failed with below error:

Tag: release-5.1.0_RC5
`TASK [azure-cloud-storage : delete files and folders from azure storage using azcopy] ***
fatal: [localhost]: FAILED! => {
    "ansible_job_id": "337217359807.5713",
    "changed": true,
    "cmd": "azcopy rm 'https://preprodall.blob.core.windows.net/sunbird-content-preprod/content-editor?se=2023-01-18T11%3A38Z&sp=rwdl&spr=https&sv=2018-11-09&sr=c&sig=lviojJgGMxgC30suWTvQ5ft94Qi61LjlR/2nIVJfH/I%3D' --recursive",
    "delta": "0:00:00.172462",
    "end": "2023-01-18 16:08:25.250392",
    "finished": 1,
    "rc": 1,
    "start": "2023-01-18 16:08:25.077930"
}

STDOUT:

INFO: Any empty folders will not be processed, because source and/or destination doesn't have full folder support

failed to perform remove command due to error: cannot start job due to error: cannot list files due to reason -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /home/vsts/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.10.1-0.20210407023846-16cf969ec1c3/azblob/zc_storage_error.go:42
===== RESPONSE ERROR (ServiceCode=ContainerNotFound) =====`

**3) Taxonamy Deployment Issue**
Taxonomy  deployment is job is failed with below error:

Tag: release-5.1.0_RC6

`TASK [sunbird-deploy : copy and template env file from swarm role] *************
failed: [localhost] (item=../../../../ansible/roles/stack-sunbird/templates/taxonomy-service_application.conf) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": "../../../../ansible/roles/stack-sunbird/templates/taxonomy-service_application.conf"
}

MSG:

AnsibleUndefinedVariable: 'dict object' has no attribute 'lp-redis-ps'`

**4)Dial service deployment Issue**
Dial deployment is job is failed with below error:

Tag: release-5.1.0_RC6

`TASK [sunbird-deploy : copy and template env file from swarm role] *************
failed: [localhost] (item=../../../../ansible/roles/stack-sunbird/templates/dial-service_application.conf) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": "../../../../ansible/roles/stack-sunbird/templates/dial-service_application.conf"
}`

**5)Assesement  deployment is job is failed with below error:**

Tag: release-5.1.0_RC6
`TASK [sunbird-deploy : copy and template env file from swarm role] *************
failed: [localhost] (item=../../../../ansible/roles/stack-sunbird/templates/assessment-service_application.conf) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": "../../../../ansible/roles/stack-sunbird/templates/assessment-service_application.conf"
}`

6)**Content Editor** deployment is job is failed with below error
Tag: release-5.1.0_RC5
`TASK [azure-cloud-storage : delete files and folders from azure storage using azcopy] ***
fatal: [localhost]: FAILED! => {
    "ansible_job_id": "337217359807.5713",
    "changed": true,
    "cmd": "azcopy rm 'https://preprodall.blob.core.windows.net/sunbird-content-preprod/content-editor?se=2023-01-18T11%3A38Z&sp=rwdl&spr=https&sv=2018-11-09&sr=c&sig=lviojJgGMxgC30suWTvQ5ft94Qi61LjlR/2nIVJfH/I%3D' --recursive",
    "delta": "0:00:00.172462",
    "end": "2023-01-18 16:08:25.250392",
    "finished": 1,
    "rc": 1,
    "start": "2023-01-18 16:08:25.077930"
}

STDOUT:

INFO: Any empty folders will not be processed, because source and/or destination doesn't have full folder support

failed to perform remove command due to error: cannot start job due to error: cannot list files due to reason -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /home/vsts/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.10.1-0.20210407023846-16cf969ec1c3/azblob/zc_storage_error.go:42
===== RESPONSE ERROR (ServiceCode=ContainerNotFound) =====
Description=The specified container does not exist.
RequestId:25babe4e-101e-0066-3128-2bd61d000000
Time:2023-01-18T10:38:25.2417796Z, Details: 
   Code: ContainerNotFound
   GET https://preprodall.blob.core.windows.net/sunbird-content-preprod?comp=list&delimiter=%2F&include=metadata&prefix=content-editor%2F&restype=container&se=2023-01-18t11%3A38z&sig=-REDACTED-&sp=rwdl&spr=https&sr=c&sv=2018-11-09&timeout=901
   User-Agent: [AzCopy/10.10.0 Azure-Storage/0.13 (go1.15; linux)]
   X-Ms-Client-Request-Id: [72c1bd53-3f04-4799-6bdd-34555be1cbc3]
   X-Ms-Version: [2019-12-12]
   --------------------------------------------------------------------------------
   RESPONSE Status: 404 The specified container does not exist.
   Access-Control-Allow-Origin: [*]
   Access-Control-Expose-Headers: [Accept-Ranges,Access-Control-Allow-Methods,Access-Control-Allow-Origin,Access-Control-Expose-Headers,Content-Encoding,Content-Length,Content-Range]
   Content-Length: [225]
   Content-Type: [application/xml]
   Date: [Wed, 18 Jan 2023 10:38:24 GMT]
   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]
   X-Ms-Client-Request-Id: [72c1bd53-3f04-4799-6bdd-34555be1cbc3]
   X-Ms-Error-Code: [ContainerNotFound]
   X-Ms-Request-Id: [25babe4e-101e-0066-3128-2bd61d000000]
   X-Ms-Version: [2019-12-12]`

7)**LernFlinkJobs deployment is job is failed with below error**
Tag : release-5.0.1_RC4
`PLAY [local] *******************************************************************

TASK [flink-jobs-deploy : template values.yaml file] ***************************
fatal: [localhost]: FAILED! => {
    "changed": false
}

MSG:

AnsibleUndefinedVariable: 'is_multidc_enabled' is undefined

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0`

**Comment by chitranshu-keshav** [comment]
@vinukumar-vs @gandham-santhosh - pls check and respond

**Reply by gandham-santhosh** [reply]
@sandeep-r1 looks like these deployments were failing as some variables were not updated. Please update the variables in common.yaml and re-deploy the services. 
Are these issues fixed now?

**Reply by chitranshu-keshav** [reply]
@sandeep-r1 awaiting your response on the above. Can you please confirm if the issue is resolved

**Comment by chitranshu-keshav** [comment]
Closing this thread as there is no response
